### DISCUSSION TITLE
How to style a collapsed `Tabs` component's `Picker`?

### DISCUSSION THREAD

**Discussion by mdoliv** [adobe/react-spectrum#5325] [Q&A] [answered]
I'm using tabs to alternate between three different search form options, and I'm rendering a `ContextualHelp` for a couple of those tabs. 

![image](https://github.com/adobe/react-spectrum/assets/22868215/021965ee-439c-4a79-bcc1-9b804e50d3ef)

This is all fine, but once they get collapsed, the `ContextualHelp` is displayed in the `Picker`'s items but with a weird positioning. 

![image](https://github.com/adobe/react-spectrum/assets/22868215/cd2a88bc-50f8-4fa2-a210-3897d25e6b3a)

I wonder if there's any way to style the `Item`s of the collapsed `Tabs` in order to not display the `ContextualHelp`s when the `Tabs` component gets collapsed.

Also, on a similar note, is there an easy way of centering the `TabList` without it collapsing when there's clearly enough space available? For example, when I do something like:

```jsx
<Tabs>
  <TabList alignSelf="center">
  {/* ... */}
  </TabList>
</Tabs>
```

It does center nicely and it works properly on large windows, but on my laptop window it gets collapsed even though there's clearly enough room to accommodate the centered `TabList`:

![image](https://github.com/adobe/react-spectrum/assets/22868215/01a6bed2-0350-47a5-a46d-be83578597dc)

**Comment by snowystinger** [comment, marked as answer]
It's not accessible to have a contextual help or anything else interactive inside a Tab. See https://w3c.github.io/aria/#tab `Children presentational: True`. I'm not sure how a keyboard user would get to that contextual help either.

I suggest finding a different way to relay the information in the contextual help. You could put one outside the tabs, much like the "add"/"remove" buttons are added here https://react-spectrum.adobe.com/react-spectrum/Tabs.html#customizing-layout

Once you remove it from there, you won't need to worry about it inside the collapsed menu. This is a good thing because in Spectrum, that icon would indicate that the item is unavailable, which is not what it seems you're trying to convey. See https://react-spectrum.adobe.com/react-spectrum/Menu.html#unavailable-items

**Reply by mdoliv** [reply]
Great, thanks for the advice!

Any insight on the centering issue? I can't seem to get it to work properly without specifying some magic values for the `width` and `max-width`. Even wrapping it in a `Flex` and giving it a `flexGrow` doesn't work because then it overgrows and stops being centered. I know I'm noobing on my CSS a bit here but help would be much appreciated.

**Reply by snowystinger** [reply]
That's a hard one, I'm not sure you can do that right now
https://codesandbox.io/s/gifted-benz-tn4xhp?file=/src/App.js is the closest I got, just delete the "flex: 1 1 auto" line, however, once it collapses, it won't un-collapse. Would need more time to think about it, or you'd need to open a PR to support it possibly.
